test/views/AgentForOrgCheckRegisteredDetailsViewSpec.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import forms.AgentForOrgCheckRegisteredDetailsFormProvider
+import models.{DraftId, NormalMode, TraderDetailsWithCountryCode}
+import play.twirl.api.HtmlFormat
+import views.behaviours.ViewBehaviours
+import views.html.AgentForOrgCheckRegisteredDetailsView
+
+class AgentForOrgCheckRegisteredDetailsViewSpec extends ViewBehaviours {
+
+  val messageKeyPrefix = "checkRegisteredDetails"
+
+  val doesNotConsentToDisclosureOfPersonalData: TraderDetailsWithCountryCode =
+    traderDetailsWithCountryCode.copy(consentToDisclosureOfPersonalData = false)
+
+  val emptyCDSFullName: TraderDetailsWithCountryCode =
+    traderDetailsWithCountryCode.copy(CDSFullName = "")
+
+  val form: AgentForOrgCheckRegisteredDetailsFormProvider =
+    app.injector.instanceOf[AgentForOrgCheckRegisteredDetailsFormProvider]
+
+  val view: AgentForOrgCheckRegisteredDetailsView = app.injector.instanceOf[AgentForOrgCheckRegisteredDetailsView]
+
+  val viewViaApply: HtmlFormat.Appendable  =
+    view(form.apply(), traderDetailsWithCountryCode, NormalMode, DraftId(1L))(fakeRequest, messages)
+  val viewViaRender: HtmlFormat.Appendable =
+    view.render(form.apply(), traderDetailsWithCountryCode, NormalMode, DraftId(1L), fakeRequest, messages)
+  val viewViaF: HtmlFormat.Appendable      =
+    view.f(form.apply(), traderDetailsWithCountryCode, NormalMode, DraftId(1L))(fakeRequest, messages)
+
+  "AgentForOrgCheckRegisteredDetailsView" - {
+    normalPage(messageKeyPrefix, "agentOrg")()
+  }
+
+  "the trader does not consent to disclosure of personal data" - {
+    val viewAlternate =
+      view(form.apply(), doesNotConsentToDisclosureOfPersonalData, NormalMode, DraftId(1L))(
+        fakeRequest,
+        messages
+      )
+
+    pageByMethodWithAssertions(
+      viewAlternate,
+      "checkRegisteredDetails.private",
+      "",
+      doesNotConsentToDisclosureOfPersonalData.EORINo
+    )() {
+      "contains the correct heading" in {
+        assertContainsMessages(
+          asDocument(viewAlternate),
+          messages("checkRegisteredDetails.private.heading", doesNotConsentToDisclosureOfPersonalData.EORINo)
+        )
+      }
+    }
```

Comment:
I'm trying to understand why pageByMethodWithAssertions instead of pageByMethod was used here. As I understand, pageByMethodWithAssertions calls the method pageByMethod which tests the title and heading so currently it seems the heading is being tested twice. Is this meant to test something else? If it is just the heading that then I think pageByMethod should be used here.

Reply:
Yeh made mistake here should have been paragraph like the Employee version of this page.